src/components/OpenWith/OpenWith.js

```diff
@@ -112,16 +121,60 @@ class OpenWith extends PureComponent<Props, State> {
      * @return {void}
      */
     componentDidMount() {
-        const { fileId }: Props = this.props;
+        this.fetchOpenWithData();
+    }
+
+    /**
+     * Fetches file extension and Open With data.
+     *
+     * @return {void}
+     */
+    fetchOpenWithData() {
+        const { fileId, extension, language }: Props = this.props;
+        if (extension) {
```

Comment:
It seems like this operation should happen separately: once in the `initialState` construction and once again in componentDidUpdate if the extension changed.

Reply:
The extension can/should only change if the file ID changes. That's why I have them grouped currently.